daemon/cluster/listen_addr.go

```diff
@@ -103,6 +88,23 @@ func (c *Cluster) resolveAdvertiseAddr(advertiseAddr, listenAddrPort string) (st
 	return systemAddr.String(), listenAddrPort, nil
 }
 
+func resolveDataPathAddr(dataPathAddr string) (string, error) {
+	// Approach:
+	// - If a data path flag is specified try to resolve the IP address.
+	if dataPathAddr != "" {
```

Comment:
I would handle the empty case right away and avoid the extra indentation here with 
```
if dataPathAddr == "" {
    return "", nil
}
```